app/domain/invoices/abacus/course_annulation_cost.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+module Invoices
+  module Abacus
+    class CourseAnnulationCost
+      # cancellation in [range] days before course start => [factor]
+      CANCELLATION_COST_FACTORS = {
+        ..9 => 1,
+        10..19 => 0.75,
+        20..30 => 0.5
+      }
+      CANCELLATION_PROCESSING_FEE = 80.0
+
+      def initialize(participation)
+        @participation = participation
+      end
+
+      def position_description_and_amount_cancelled
+        days_until_start = (course_start_date - cancelled_at).to_i
+
+        range = CANCELLATION_COST_FACTORS.keys.find { |range| range.include?(days_until_start) }
+        if range && @participation.price
```

Comment:
Wie genau sollen wir damit umgehen falls `@participation.price = nil` ist?
Zuvor war das noch nicht gehandled aber da das jetzt bei der View verwendet wird müssen wir das schon irgendwie abdecken.. Die Spalte in der DB ist ja nullable
Hast du da eine Meinung dazu? @codez

Reply:
In diesem Fall würde ich als Betrag ebenfalls 0.00 ausgeben. Wer in den Genuss eines kostenlosen Kurses kommt, muss wohl auch keine Annulationskosten übernehmen.